WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -121,7 +125,13 @@ private bool TryGetBestKnownByRecipientPockets(Pocket[] knownByRecipientPockets,
 			}
 		}
 
-		foreach (var p in bestPockets.OrderBy(x => x.Amount).ThenByDescending(x => x.Labels.Count()).ToImmutableArray())
+		// It can happen there unnecessary selected pockets, so remove the ones that are not needed.
+		// Use Except to make sure the private and semi private pockets never get removed.
+		// Example for an over selection:
+		// Privacy ordered pockets: [A - 3 BTC] [B - 1 BTC] [C - 2 BTC] (A is the best for privacy, C is the worst)
+		// Target amount is 4.5 BTC so to algorithm will select all because it happened in privacy order.
+		// But B is unnecessary because A and C can cover the case, so remove it.
```

Comment:
```suggestion
		// It can happen that there are unnecessary selected pockets, so remove the ones that are not needed.
		// Use Except to make sure the private and semi private pockets never get removed.
		// Example for an over selection:
		// Privacy ordered pockets: [A - 3 BTC] [B - 1 BTC] [C - 2 BTC] (A is the best for privacy, C is the worst)
		// Target amount is 4.5 BTC so the algorithm will select all because it happened in privacy order.
		// But B is unnecessary because A and C can cover the case, so remove it.
```